test/metric/dimension/provider.go

```diff
@@ -16,11 +16,11 @@ type ExpectedDimensionValue struct {
 }
 
 func (d *ExpectedDimensionValue) IsKnown() bool {
-	return d.Value == nil
+	return d.Value != nil
```

Comment:
This file contains the fix. The other stuff is cleanup / improving logging in case we have an issue with this test in the future.